requirements.txt

```diff
@@ -1,4 +1,4 @@
-numpy>=1.18.0
+numpy>=1.21.0
```

Comment:
This is a quiet but potentially important change. At the very least we will need to properly mention this in the changelog. And we definitely have to think about the potential implications, because if some other library on some version on some system as a requirement `numpy <= 1.20`, then everything breaks.

As I understand, this is the minimum version required for `np.typing.NDArray`?

Reply:
Yes, you are correct. This is required for `np.typing.NDArray`.
I will add a comment on the PR explaining this

One of the limitations of < 1.21 is no python 3.6 version exists but if we are removing python 3.6 in the next release then this is fine. 
Do you know why 1.20 causes issues?